src/operator/tensor/square_sum-inl.h

```diff
@@ -28,13 +28,16 @@ inline bool SquareSumForwardInferStorageType(const nnvm::NodeAttrs& attrs,
                                              std::vector<int>* out_attrs) {
   CHECK_EQ(in_attrs->size(), 1U);
   CHECK_EQ(out_attrs->size(), 1U);
-  CHECK_EQ((*in_attrs)[0], kRowSparseStorage)
-    << "_square_sum only supports row-sparse ndarray as input";
   const ReduceAxesParam& param = nnvm::get<ReduceAxesParam>(attrs.parsed);
-  if (param.axis[0] == 1 && param.keepdims) {  // sum per row and keep dims
-    STORAGE_TYPE_ASSIGN_CHECK(*out_attrs, 0, kRowSparseStorage);
-  } else {
-    STORAGE_TYPE_ASSIGN_CHECK(*out_attrs, 0, kDefaultStorage);
+  if (in_attrs->at(0) == kRowSparseStorage) {  // current impl
+    if (param.axis[0] == 1 && param.keepdims) {  // sum per row and keep dims
+      STORAGE_TYPE_ASSIGN_CHECK(*out_attrs, 0, kRowSparseStorage);
+    } else {
+      STORAGE_TYPE_ASSIGN_CHECK(*out_attrs, 0, kDefaultStorage);
+    }
+  } else {  // fallback
+      STORAGE_TYPE_ASSIGN_CHECK(*in_attrs, 0, kDefaultStorage);
+      STORAGE_TYPE_ASSIGN_CHECK(*out_attrs, 0, kDefaultStorage);
```

Comment:
The purpose of fallback is to use FCompute, although there isn't one for this op. There shouldn't be any attrs assigned kRowSparseStorage/kCSRStorage under `else` condition. If there are any, it should fail. That's why STORAGE_TYPE_ASSIGN_CHECK is used. When it reaches FComputeEx, the in_stype should not be undefined (otherwise, it means infer storage failure in the infer storage pass), just like all the dtypes should not be -1 when the program reaches FCompute.